module/os/freebsd/zfs/zio_crypt.c

```diff
@@ -1070,6 +1070,19 @@ zio_crypt_do_objset_hmacs(zio_crypt_key_t *key, void *data, uint_t datalen,
 
 	bcopy(raw_portable_mac, portable_mac, ZIO_OBJSET_MAC_LEN);
 
+	/*
+	 * Add in the non-portable os_flags. This is necessary here as we check
+	 * whether OBJSET_FLAG_USERACCOUNTING_COMPLETE is set in order to
+	 * decide if the local_mac should be zeroed out.
+	 */
+	intval = osp->os_flags;
+	if (should_bswap)
+		intval = BSWAP_64(intval);
+	intval &= ~OBJSET_CRYPT_PORTABLE_FLAGS_MASK;
+	/* CONSTCOND */
+	if (!ZFS_HOST_BYTEORDER)
+		intval = BSWAP_64(intval);
+
```

Comment:
I think simply moving the logic from below up here is not sufficient. We are trying to check if the flag is set. This should be done by checking the bit after swapping the osp into our native byteorder. Here, however we might possibly switch it again, I think? Check my logic here.

It might be cleaner to just do the checking logic separately from the MAC calculation (even if we do end up redoing some work).

Reply:
I see what you mean. I will opt for the cleaner option, redoing the calculation.